MobiusCore/Source/EffectHandlers/EffectHandler.swift

```diff
@@ -17,15 +17,16 @@
 // specific language governing permissions and limitations
 // under the License.
 
-/// This protocol defines the contract for an Effect Handler which takes `Effect`s as input, and produces `Event`s as
-/// output.
+/// This protocol defines the contract for an Effect Handler which takes `EffectParameters` as input, and produces
+/// `Event`s as output.
 ///
-/// For each incoming `Effect`, zero or more `Event`s can be sent as output using `callback.send(Event)`. When an
-/// `Effect` has been completely handled, i.e. that effect will not result in any more events, call `callback.end()`.
+/// For each incoming effect parameter, zero or more `Event`s can be sent as output using `callback.send(Event)`. When
+/// an the effect parameters have been completely handled, i.e. that effect will not result in any more events, call
```

Comment:
This doesn't read well, IMO. In the [Java wiki](https://github.com/spotify/mobius/wiki/Concepts#effect), we talk about effect descriptions and effect execution as separate things. The effect description is the combination of an effect type and values for its parameters. I guess `EffectParameters` to be strict would be the types and names of the parameters (to be analogous with the definition of method parameters), not necessarily their values. In this case, it's parameters with assigned values. Arguments, maybe? Not exactly the same either.

I sort of think that Effect is clearer or at least more precise than EffectParameters. If nothing else, then at least in this paragraph, which should state that "for each incoming effect, zero or more.." rather than "for each incoming effect parameter, ...". The latter sounds like you would potentially send one event for each of the parameters of an effect. 

I do think effect parameters makes sense as a name for the 'names + types' of the data associated with the enum value. But I don't think it makes great sense for the 'names + types + values'. It's not completely unreasonable to use the name `EffectParameters` as a type alias because it's fine to fuzz the distinction about whether there are values or not, but it's definitely not the 'parameters' that the effect handler is handling, but rather the 'effect'.

Reply:
I have clarified the docs.
I agree that the name is not ideal, but I think it's better than overloading the term `Effect`.

Reply:
That's probably true - but the docs actually refer to handling the `Effect` as a whole, not the individual parameters/arguments. I tried to clarify what I mean above.

I'm slightly leaning towards thinking `EffectArguments` is better than `EffectParameters`, but that's not a huge deal, it'll be clear what it means, I think.